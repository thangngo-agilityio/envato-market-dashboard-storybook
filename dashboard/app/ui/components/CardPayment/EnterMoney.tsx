import { memo } from 'react';
import { Box, Button, Flex, Input, Text } from '@chakra-ui/react';
import { Control, Controller } from 'react-hook-form';

// Types
import { TTransfer } from '.';

// Utils
import { formatAmountNumber } from '@/lib/utils';
import { AUTH_SCHEMA } from '@/lib/constants';

export type TEnterMoneyProps = {
  control: Control<TTransfer>;
  isDisabled?: boolean;
};

const EnterMoneyComponent = ({
  control,
  isDisabled = false,
}: TEnterMoneyProps): JSX.Element => (
  <>
    <Box
      border="1px solid"
      borderColor="border.secondary"
      p={4}
      mt={5}
      borderRadius="lg"
    >
      <Text color="text.secondary"> Enter amount </Text>
      <Flex direction="row" alignItems="center">
        <Text color="text.primary" fontSize="2xl" fontWeight="bold">
          $
        </Text>
        <Controller
          control={control}
          rules={AUTH_SCHEMA.TRANSFER_AMOUNT}
          name="amount"
          defaultValue=""
          render={({ field: { value, onChange } }) => {
            const handleChange = (
              event: React.ChangeEvent<HTMLInputElement>,
            ) => {
              const value: string = event.target.value;

              if (isNaN(+value.replaceAll(',', ''))) return;

              // Remove non-numeric characters and leading zeros
              const sanitizedValue = formatAmountNumber(value);

              onChange(sanitizedValue);
            };

            return (
              <Input
                variant="authentication"
                type="text"
                _dark={{
                  border: 'none',
                }}
                placeholder="0.00"
                sx={{ border: 'none', padding: 0 }}
                color="text.primary"
                fontWeight="bold"
                fontSize="2xl"
                ml={2}
                value={value}
                name="amount"
                onChange={handleChange}
                autoComplete="off"
                position="static"
              />
            );
          }}
        />
      </Flex>
    </Box>

    <Button
      aria-label="btn-send-money"
      mt={4}
      colorScheme="primary"
      bg="primary.300"
      fontWeight="bold"
      type="submit"
      isDisabled={isDisabled}
    >
      Send Money
    </Button>
  </>
);

const EnterMoney = memo(EnterMoneyComponent);
export default EnterMoney;
