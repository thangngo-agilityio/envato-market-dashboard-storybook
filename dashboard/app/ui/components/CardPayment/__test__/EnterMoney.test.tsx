import '@testing-library/jest-dom';
import { Control, useForm } from 'react-hook-form';

// Components
import EnterMoney from '../EnterMoney';
import { TTransfer } from '..';

const { result } = renderHook(() => useForm<TTransfer>());

const setup = () => {
  const control: Control<TTransfer> = result.current.control;
  const props = {
    control,
    onUploadError: jest.fn(),
  };

  return render(<EnterMoney {...props} />);
};

describe('CardPayment test cases', () => {
  test('CardPayment component renders correctly', () => {
    const { container } = setup();

    expect(container).toMatchSnapshot();
  });
});
