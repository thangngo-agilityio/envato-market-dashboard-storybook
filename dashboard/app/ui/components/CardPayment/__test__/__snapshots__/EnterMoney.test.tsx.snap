// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardPayment test cases CardPayment component renders correctly 1`] = `
<div>
  <div
    class="css-1hlzswh"
  >
    <p
      class="chakra-text css-1ki0mgu"
    >
       Enter amount 
    </p>
    <div
      class="css-u4p24i"
    >
      <p
        class="chakra-text css-4xj0bk"
      >
        $
      </p>
      <input
        autocomplete="off"
        class="chakra-input css-cz5fpu"
        name="amount"
        placeholder="0.00"
        type="text"
        value=""
      />
    </div>
  </div>
  <button
    aria-label="btn-send-money"
    class="chakra-button css-1mu22hy"
    type="submit"
  >
    Send Money
  </button>
  <span
    hidden=""
    id="__chakra_env"
  />
</div>
`;
